src/context/filter.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{
+    Context, RedisModuleCommandFilter, RedisModuleCommandFilterCtx, RedisModuleString,
+    RedisModule_CommandFilterArgDelete, RedisModule_CommandFilterArgGet,
+    RedisModule_CommandFilterArgInsert, RedisModule_CommandFilterArgReplace,
+    RedisModule_CommandFilterArgsCount, RedisModule_CommandFilterGetClientId,
+    RedisModule_RegisterCommandFilter, RedisModule_UnregisterCommandFilter, ValkeyString,
+};
+use std::ffi::c_int;
+use std::str::Utf8Error;
+
+#[derive(Debug, Clone, Copy)]
+pub struct CommandFilter {
+    inner: *mut RedisModuleCommandFilter,
+}
+
+// otherwise you get error:     cannot be shared between threads safely
+unsafe impl Send for CommandFilter {}
+unsafe impl Sync for CommandFilter {}
+
+impl CommandFilter {
+    pub fn is_null(&self) -> bool {
+        self.inner.is_null()
+    }
+
+    pub fn args_count(ctx: *mut RedisModuleCommandFilterCtx) -> c_int {
```

Comment:
Just two suggestions on the updated code:

1. For all the functions implemented (`impl`) on the CommandFilter struct that accept `*mut RedisModuleCommandFilterCtx`, you could change the fn signature to accept `&self` or `&mut self` depending on the function. See: https://github.com/valkey-io/valkeymodule-rs/blob/main/src/defrag.rs#L21
2. You can implement a `new` constructor to help create `CommandFilter`. See: https://github.com/valkey-io/valkeymodule-rs/blob/main/src/defrag.rs#L12